src/Interactive/EditorFeatures/Core/Extensibility/Interactive/InteractiveEvaluator.cs

```diff
@@ -323,10 +331,14 @@ private void AddSubmission(ITextView textView, ITextBuffer subjectBuffer)
             }
 
             // project for the new submission:
-            project = CreateSubmissionProject(solution);
+            project = CreateSubmissionProject(solution, languageName);
 
             // Keep track of this buffer so we can freeze the classifications for it in the future.
-            _submissionBuffers.Add(ValueTuple.Create(textView, subjectBuffer));
+            var viewAndBuffer = ValueTuple.Create(textView, subjectBuffer);
+            if (!_submissionBuffers.Contains(viewAndBuffer))
```

Comment:
Since order doesn't appear to matter, perhaps ```_submissionBuffers``` should be a set, rather than a list.